DEVELOPER_GUIDE.md

```diff
@@ -422,6 +423,77 @@ Output plugins are defined in [flb_output.h](https://github.com/fluent/fluent-bi
 
 The [stdout plugin](plugins/out_stdout) is very simple; review its code to understand how output plugins work.
 
+#### Config Maps
+
+Config maps are an improvement to the previous fluent-bit exposed API that was used by plugins to read configuration values. The new config maps feature warns the user in case of getting an unknown configuration key and reduces risk of  bad configuration due to typos or deprecated properties names. They are also important  so that in the future dynamic reload support can be implemented.
+
+There are various types of supported configuration types. Full list available [here](https://github.com/fluent/fluent-bit/blob/master/include/fluent-bit/flb_config_map.h#L29). The most used ones are :
+
+```
+FLB_CONFIG_MAP_INT - Represents integer data type.
+FLB_CONFIG_MAP_BOOL - Represents boolean data type.
+FLB_CONFIG_MAP_DOUBLE - Represents a double.
+FLB_CONFIG_MAP_SIZE - Converts string size to integer
+FLB_CONFIG_MAP_STR - Represents string data type.
+FLB_CONFIG_MAP_CLIST - Represents comma separated list of strings.
+FLB_CONFIG_MAP_SLIST - Represents empty space separated list of strings.
+
+```
+A config map expects certain public fields at registration. They are 
+
```

Comment:
add more descriptions to each value, a developer who aims to learn from this document needs real explanations